packages/@aws-cdk-testing/cli-integ/lib/xpmutex.ts

```diff
@@ -0,0 +1,218 @@
+import { watch, promises as fs, mkdirSync } from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+export class XpMutexPool {
+  public static fromDirectory(directory: string) {
+    mkdirSync(directory, { recursive: true });
+    return new XpMutexPool(directory);
+  }
+
+  public static fromName(name: string) {
+    return XpMutexPool.fromDirectory(path.join(os.tmpdir(), name));
+  }
+
+  private readonly queuedResolvers = new Set<() => void>();
```

Comment:
Why a `Set` and not a `Queue` (based on name and the FIFO comment from a previous file)?

Reply:
Hm. There are definitely no FIFO guarantees. Where did you read that?